### PR TITLE
switch all queries to use react-query

### PR DESCRIPTION
**Pull request by KillerByte**
For now no UI elements have been chaged. For suggested future work, it would make sense to pass the `QueryStatus` objects related to a stat directly to the `StatsBox`. The `StatsBox` can then be modified to show live reload, errors, or force reload.

`useContext` for SNX and sUSD were replaced with `useSNXInfo` and `useSUSDInfo`, which performs the necessary queries to populate the data. since `react-query` maintains its own internal cache, there is no need to use context.

The default auto-refresh period used by `react-query` might be a bit to aggressive for the purposes of this dashboard, particularly if an API goes down or starts producing exceptions. Please advise on if this should be adjusted.

### CODE REVIEW

**Comment by dvd-schwrtz** [comment]
for now, let's have the queries with 0 retries, until we are ready to pass the query status to the StatsBox and implement the retry / refetch logic UI with those buttons. I have some stuff to do over the next several hours but I will try and send you some designs for that later today. We have a new contributor, @jchiaramonte7, who is familiar with react-qery that will be reviewing this PR in the meantime :)

**Review by JChiaramonte7 — COMMENTED** [review]
Looks good to me. Everything seems to be working as intended. 

Just some small comments and you also have an empty `queries/network/index.ts`

File: utils/dataMapping.ts [hunk 1]
```diff
@@ -11,3 +11,16 @@ export const getSUSDHoldersName = (id: string): string => {
 	const address = id.split('-')[0];
 	return KNOWN_ADDRESSES[address] ?? 'unknown';
 };
+
+export function periodToDays(period: string): number {
```
Comment (dvd-schwrtz) [inline utils/dataMapping.ts:15]: ChartPeriod type instead of string

File: queries/yield-farming/useCurveContractInfoQuery.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	);
+	const curveGaugeControllerContract = new ethers.Contract(
+		curveGaugeController.address,
+		// @ts-ignore
+		curveGaugeController.abi,
+		provider
+	);
+
+	return useQuery<CurveContractInfo, string>(QUERY_KEYS.YieldFarming.CurveInfo, async () => {
+		const r = await Promise.all([
```
Comment (dvd-schwrtz) [inline queries/yield-farming/useCurveContractInfoQuery.ts:47]: I don't hate this but I also don't love it when we have single letter variables. just a nitpick. I use them for loops often. and this is just a response you map right away so its not confusing but in general I think being more verbose is nicer for those reading the code whenever possible

**Comment by dvd-schwrtz** [comment]
great job!

**Review by JChiaramonte7 — APPROVED** [review]

**Pull request merged**